### PR TITLE
Text track integration

### PR DESCRIPTION
**Pull request by aczs**
Summary: Text track support added
Type: Feature
Test Plan: UT, CT, YT Cert Test
Jira: RIALTO-591

### CODE REVIEW

**Review by skywojciechowskim — APPROVED** [review]

File: media/public/include/IMediaPipeline.h [hunk 1]
```diff
@@ -1299,24 +1299,42 @@ class IMediaPipeline
...
+     * @param[in] mute   Desired mute state, true=muted, false=not muted
      *
      * @retval true on success false otherwise
      */
-    virtual bool setMute(bool mute) = 0;
+    virtual bool setMute(int32_t sourceId, bool mute) = 0;
 
     /**
-     * @brief Get current mute status of the pipeline
+     * @brief Get current mute status of the media source
```
Comment (skyfroxl) [inline media/public/include/IMediaPipeline.h:1312]: Add a comment that we only support set/get mute for Audio and Subtitles

File: media/server/main/include/MediaPipelineServerInternal.h [hunk 2]
```diff
@@ -475,7 +479,25 @@ class MediaPipelineServerInternal : public IMediaPipelineServerInternal, public
...
+     * @param[in] textTrackIdentifier Text track identifier of subtitle stream
+     *
+     * @retval true on success false otherwise
+     */
+    bool setTextTrackIdentifierInternal(const std::string &textTrackIdentifier);
+
+    /**
+     * @brief Get Text Track Identifier
+     *
+     * @param[in] textTrackIdentifier Text track identifier of subtitle stream
```
Comment (skyfroxl) [inline media/server/main/include/MediaPipelineServerInternal.h:496]: out

File: media/server/main/source/SharedMemoryBuffer.cpp [hunk 3]
```diff
@@ -255,18 +257,26 @@ bool SharedMemoryBuffer::clearData(MediaPlaybackType playbackType, int id, const
...
     if (MediaSourceType::AUDIO == mediaSourceType)
     {
         std::uint8_t *audioData = partitionDataPtr + partition->dataBufferVideoLen;
         memset(audioData, 0x00, partition->dataBufferAudioLen);
         return true;
     }
-    if (MediaSourceType::VIDEO == mediaSourceType)
+    if (MediaSourceType::SUBTITLE == mediaSourceType)
     {
-        std::uint8_t *videoData = partitionDataPtr;
```
Comment (skyfroxl) [inline media/server/main/source/SharedMemoryBuffer.cpp:266]: Reminder to update shm documentation https://wiki.rdkcentral.com/display/ASP/Rialto+Playback+Design

File: media/server/main/source/TextTrackSession.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    static TextTrackSessionFactory factory;
+    return factory;
+}
+
+std::unique_ptr<ITextTrackSession> TextTrackSessionFactory::createTextTrackSession(const std::string &display) const
+{
+    return std::make_unique<TextTrackSession>(display, ITextTrackAccessorFactory::getFactory());
+}
+
+TextTrackSession::TextTrackSession(const std::string &displayName,
```
Comment (skyfroxl) [inline media/server/main/source/TextTrackSession.cpp:38]: Any issues with calling thunder on calling thread from gstreamer?

File: media/server/main/source/MediaPipelineServerInternal.cpp [hunk 5]
```diff
@@ -903,7 +913,66 @@ bool MediaPipelineServerInternal::getMuteInternal(bool &mute)
...
+    return result;
+}
+
+bool MediaPipelineServerInternal::getTextTrackIdentifierInternal(std::string &textTrackIdentifier)
+{
+    RIALTO_SERVER_LOG_DEBUG("entry:");
+
+    if (!m_gstPlayer)
+    {
+        RIALTO_SERVER_LOG_ERROR("Failed to get mute - Gstreamer player has not been loaded");
```
Comment (skyfroxl) [inline media/server/main/source/MediaPipelineServerInternal.cpp:971]: Update comment

File: proto/mediapipelinemodule.proto [hunk 6]
```diff
@@ -452,11 +454,43 @@ message SetMuteResponse {
...
+ */
+message SetTextTrackIdentifierRequest {
+    optional int32 session_id = 1 [default = -1];
+    optional string text_track_identifier = 2;
+}
+message SetTextTrackIdentifierResponse {
+}
+
+/**
+ * @fn void getTextTrackIdentifier(std::string &textTrackIdentifier)
```
Comment (skyfroxl) [inline proto/mediapipelinemodule.proto:480]: string return

**Review by skyfroxl — APPROVED** [review]
None of these comments are blockers!

**Pull request merged**